source/Calamari.Shared/Commands/ApplyDeltaCommand.cs

```diff
@@ -56,7 +56,14 @@ public override int Execute(string[] commandLineArguments)
             try
             {
                 ValidateParameters(out basisFilePath, out deltaFilePath, out newFilePath);
-                packageJournal.ApplyRetention(PackageStore.GetPackagesDirectory());
+                try
+                {
+                    packageJournal.ApplyRetention(PackageStore.GetPackagesDirectory());
+                }
+                catch (Exception ex)
```

Comment:
I don't understand the reason behind the addition of this `try/catch`. There are other places that `packageJournal.ApplyRetention` is called (eg in the [MavenPackageDownloader](https://github.com/OctopusDeploy/Calamari/blob/master/source/Calamari.Shared/Integration/Packages/Download/MavenPackageDownloader.cs#L137))... would they need a similar guard?

Reply:
The other usages use a packages directory variable which has already been determined - this is the only one which calls `PackageStore.GetPackagesDirectory()`, which can throw an exception if an environment variable is missing.

Reply:
Ah okay, it's the `GetPackagesDirectory()` you're guarding rather than `ApplyRetention()`?

I assume you're talking about the `TentacleHome` variable... was that not set in the E2E tests or something?

Reply:
No, that didn't actually cause the failure for the E2E test - this is just me being a bit paranoid in making sure there's _nothing_ in retention that could fail TBH.  It may not be necessary.

The actual failure was caused by `new PackageIdentity(variables)` [here](https://github.com/OctopusDeploy/Calamari/blob/2dcc0c53b32f89f22f5213812166dba645cea387/source/Calamari/Deployment/PackageRetention/Model/Journal.cs#L49).